main/channel.c

```diff
@@ -1278,67 +1308,92 @@ struct ast_channel *ast_channel_callback(ao2_callback_data_fn *cb_fn, void *arg,
 static int ast_channel_by_name_cb(void *obj, void *arg, void *data, int flags)
 {
 	struct ast_channel *chan = obj;
-	const char *name = arg;
-	size_t name_len = *(size_t *) data;
+	struct ast_channel_callback_data *callback_data = data;
+        const char *name = arg;
+	size_t name_len = *(size_t *) callback_data->search_length;
 	int ret = CMP_MATCH;
 
 	if (ast_strlen_zero(name)) {
 		ast_log(LOG_ERROR, "BUG! Must supply a channel name or partial name to match!\n");
 		return CMP_STOP;
 	}
 
-	ast_channel_lock(chan);
+	if (ast_channel_trylock(chan) != 0) {
+		ast_verb(2, "ast_channel_by_name_cb: Deadlock-Avoid RETRY NEEDED for: %s\n", name);
```

Comment:
This should NOT be a verbose and there should NOT be a function name prefix in the log message.

Reply:
I remembered that after I committed. I was using verb for testing so I didn't have to core debug to see it.  That'll get fixed.